cardano-db-sync/src/Cardano/DbSync/LedgerState.hs

```diff
@@ -408,7 +421,27 @@ saveCurrentLedgerState env ledger mEpochNo = do
           logInfo (leTrace env) $
             mconcat
               ["File ", Text.pack file, " exists"]
-        else do
+        else
+          atomically $ writeTBQueue (leStateWriteQueue env) (file, lState)
+
+runLedgerStateWriteThread :: Trace IO Text -> LedgerEnv -> IO ()
+runLedgerStateWriteThread tracer lenv =
+  case lenv of
+    HasLedger le -> ledgerStateWriteLoop tracer (leStateWriteQueue le) (configCodec $ getTopLevelconfigHasLedger le)
+    NoLedger _ -> forever $ threadDelay 600000000 -- 10 minutes
```

Comment:
Out of interest how come we wait 10mins when there is no ledger? (don't actually know the ledger state part of the code yet)

Reply:
This is one branch of a `race_` . If there is no ledger, there is no ledger file to write. If we return immediately, the `race_` will terminate the other thread.